torch/csrc/multiprocessing/init.cpp

```diff
@@ -0,0 +1,59 @@
+#include <torch/csrc/python_headers.h>
+#include <torch/csrc/utils/object_ptr.h>
+#include <torch/csrc/utils/pybind.h>
+
+#include <iostream>
+#include <stdexcept>
+
+#if __linux__
+#include <sys/prctl.h>
+#endif
+
+#define SYSASSERT(rv, ...)                                                 \
+  if ((rv) < 0) {                                                          \
+    throw std::system_error(errno, std::system_category(), ##__VA_ARGS__); \
+  }
+
+namespace torch {
+namespace multiprocessing {
+
+namespace {
+
+PyObject* multiprocessing_init(PyObject* _unused) {
+  auto multiprocessing_module =
+      THPObjectPtr(PyImport_ImportModule("torch.multiprocessing"));
+  if (!multiprocessing_module) {
+    throw python_error();
+  }
+
+  auto module = py::handle(multiprocessing_module).cast<py::module>();
+
+  module.def("_prctl_pr_set_pdeathsig", [](int signal) {
+#if __linux__
+    auto rv = prctl(PR_SET_PDEATHSIG, signal);
+    SYSASSERT(rv, "prctl");
```

Comment:
rv is an int right? What are you asserting here?

Reply:
That the syscall didn't fail.

Reply:
Oh, i miss read that as ASSERT